cmd/contour/serve.go

```diff
@@ -234,24 +234,39 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 
 	// step 4. register our resource event handler with the k8s informers,
 	// using the SyncList to keep track of what to sync later.
-	var informerSyncList k8s.InformerSyncList
+	informerSyncList := k8s.InformerSyncList{
+		Clients: clients,
+	}
```

Comment:
I think that adding the errors from `InformOnResources` makes this setup a fair bit harder to read, and it's already pretty complicated.

What do you think about an approach like this:
```

resources, err := clients.ServerGroupsAndResources()
// error check ...

informerSyncList := k8s.InformerSyncList {
    ServerResources: resources,
    UnsupportedResources: []metav1.GroupVersionResource{},
}

// Then create informers, but the sync list will internally filter and track unsupported resources.
informerSyncList.InformOnResources(clusterInformerFactory, dynamicHandler, k8s.ServiceAPIResources()...)

for _, r := range informerSyncList.UnsupportedResources {
    // Assuming the GVR formats into a field properly ...
    log.WithField("resource", r).Info("skipping watch on unsupported resource")
}
```

Reply:
IIUC, the main use case for checking the server resources is to automatically enable the Ingress v1 APIs. In that case, maybe a pattern more like this could work:

```Go
serverResources, err := clients.ServerGroupsAndResources()

if serverResources.Supports(k8s.IngressV1Resources()) {
    // Do the stuff ...
}
```

In this approach `Supports()` would return true if all the GVRs in the list were present on the server.